app/controllers/application_controller.rb

```diff
@@ -28,12 +28,20 @@ def after_sign_in_path_for(user)
       edit_profile_path
     elsif request.referrer.present? && request.referrer != new_user_session_url
       request.referrer
-    elsif user.admin?
-      admin_events_path
-    elsif user.proposals.any? #speaker
-      proposals_path
     else
-      root_path
+      if current_event
+        if event_staff?(current_event)
+          event_staff_path(current_event)
+        else
+          event_path(current_event)
+        end
+      else
+        if user.admin?
+          admin_events_path
+        else
+          root_path
+        end
+      end
```

Comment:
I try to avoid nested if/elses if at all possible. We can simplify this a bit by putting the eslifs back in. event_staff? includes a check for whether the passed-in event exists, so eslif event_staff?(current_event) should be enough. Helps keep things readable.

I like the rule for going to the event page if current_event exists. In terms of priority, I stick it into the last elsif before the default root_path rule.